earth2studio/data/goes.py

```diff
@@ -0,0 +1,659 @@
+                )
+
+        # Set up S3 filesystem
+        try:
+            loop = asyncio.get_running_loop()
+            loop.run_until_complete(self._async_init())
+        except RuntimeError:
+            self.fs = None
+
+    async def _async_init(self) -> None:
+        """Async initialization of S3 filesystem"""
+        self.fs = s3fs.S3FileSystem(anon=True, client_kwargs={}, asynchronous=True)
+
+    def __call__(
+        self,
+        time: datetime | list[datetime] | TimeArray,
+        variable: str | list[str] | VariableArray,
+    ) -> xr.DataArray:
+        """Function to get data
+
+        Parameters
+        ----------
+        time : datetime | list[datetime] | TimeArray
+            Timestamps to return data for (UTC).
+        variable : str | list[str] | VariableArray
+            String, list of strings or array of strings that refer to variables to
+            return. Must be in the GOES lexicon.
+
+        Returns
+        -------
+        xr.DataArray
+            Data array containing the requested GOES data
+        """
+        nest_asyncio.apply()  # Patch asyncio to work in notebooks
+        try:
+            loop = asyncio.get_event_loop()
+        except RuntimeError:
+            # If no event loop exists, create one
+            loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(loop)
+
+        if self.fs is None:
+            loop.run_until_complete(self._async_init())
+
+        xr_array = loop.run_until_complete(
+            asyncio.wait_for(self.fetch(time, variable), timeout=self._async_timeout)
+        )
+
+        # Delete cache if needed
+        if not self._cache:
+            shutil.rmtree(self.cache, ignore_errors=True)
+
+        return xr_array
+
+    async def fetch(
+        self,
+        time: datetime | list[datetime] | TimeArray,
+        variable: str | list[str] | VariableArray,
+    ) -> xr.DataArray:
+        """Async function to get data
+
+        Parameters
+        ----------
+        time : datetime | list[datetime] | TimeArray
+            Timestamps to return data for
+        variable : str | list[str] | VariableArray
+            Variables to return using standardized names
+
+        Returns
+        -------
+        xr.DataArray
+            GOES data array
+        """
+        if self.fs is None:
+            raise ValueError(
+                "File store is not initialized! If you are calling this \
+            function directly make sure the data source is initialized inside the async \
+            loop!"
+            )
+
+        time, variable = prep_data_inputs(time, variable)
+        # Create cache dir if doesn't exist
+        pathlib.Path(self.cache).mkdir(parents=True, exist_ok=True)
+
+        # Make sure input time is valid
+        self._validate_time(time)
+
+        # https://filesystem-spec.readthedocs.io/en/latest/async.html#using-from-async
+        if isinstance(self.fs, s3fs.S3FileSystem):
```

Comment:
I dont think this check is needed, there is just the S3 bucket at the moment or could there be other sources?